packages/x-charts-pro/src/Heatmap/Heatmap.tsx

```diff
@@ -203,16 +228,25 @@ const Heatmap = React.forwardRef(function Heatmap(
       onAxisClick={onAxisClick}
       plugins={HEATMAP_PLUGINS}
     >
-      <ChartsSurface ref={ref} sx={sx}>
-        <g clipPath={`url(#${clipPathId})`}>
-          <HeatmapPlot slots={slots} slotProps={slotProps} />
-          <ChartsOverlay loading={loading} slots={slots} slotProps={slotProps} />
-        </g>
-        <ChartsAxis slots={slots} slotProps={slotProps} />
-        <ChartsClipPath id={clipPathId} />
-        {children}
-      </ChartsSurface>
-      {!loading && <Tooltip {...slotProps?.tooltip} />}
+      <ChartsWrapper {...chartsWrapperProps}>
+        {!hideLegend && (
+          <ChartsLegend
+            slots={{ legend: slots?.legend ?? ContinuousColorLegend }}
+            slotProps={{ legend: { labelPosition: 'extremes', ...slotProps?.legend } }}
+            sx={{ alignSelf: 'stretch' }}
```

Comment:
Should we default to middle alignment? It looks better IMO

Reply:
That's the best way I found to have something that looks ok on both directions without having to take care of it.

I tried something else:

```jsx
sx={slotProps?.legend?.direction === 'vertical' ? { height: 150 } : { width: '50%' }}
```

I need to specify the width/height because otherwise the gradient bar shrinks to a small size.

And I can not use percentage for the height. Probably because the chart wrapper does not have a defined height by itself